src/lib/operators/insert.cpp

```diff
@@ -100,115 +124,163 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
 
   _target_table = StorageManager::get().get_table(_target_table_name);
 
+  // partitioning
+  std::shared_ptr<const AbstractPartitionSchema> target_partition_schema = _target_table->get_partition_schema();
+  std::map<RowID, PartitionID> target_partition_mapping =
+      target_partition_schema->get_mapping_to_partitions(_input_table_left());
```

Comment:
This seems to assume that a partitioning schema always exists. That is fine with me, but it is inconsistent with the import, which could generate tables without a partitioning schema.

Reply:
Tables default to NullPartitionSchema